anvil/src/eth/fees.rs

```diff
@@ -71,9 +99,49 @@ impl FeeManager {
 
     /// Returns the current base fee
     pub fn set_base_fee(&self, fee: U256) {
+        trace!(target: "backend::fees", "updated base fee {:?}", fee);
         let mut base = self.base_fee.write();
         *base = fee;
     }
+
+    /// Calculates the base fee for the next block
+    pub fn get_next_block_base_fee_per_gas(
+        &self,
+        gas_used: U256,
+        gas_limit: U256,
+        last_fee_per_gas: U256,
+    ) -> u64 {
+        calc_next_base_fee(gas_used, gas_limit, last_fee_per_gas, *self.elasticity.read())
+    }
+}
+
+/// Calculates the base fee for the next block based on the given block parameters
+pub fn calc_next_base_fee(
+    gas_used: U256,
+    gas_limit: U256,
+    last_fee_per_gas: U256,
+    elasticity: f64,
+) -> u64 {
+    let gas_used = gas_used.as_u64() as f64;
+    let gas_limit = gas_limit.as_u64() as f64;
+    let gas_target = gas_limit / elasticity;
+    let last_gas_used = gas_used / (gas_target * elasticity);
```

Comment:
can this have a rounding error?

Reply:
possible, but negligible I think

but could replace this with a decimal type in a followup?